microprofile/scheduling/src/main/java/io/helidon/microprofile/scheduling/SchedulingCdiExtension.java

```diff
@@ -0,0 +1,153 @@
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.AnnotatedMethod;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.DeploymentException;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
+import javax.enterprise.inject.spi.ProcessManagedBean;
+import javax.enterprise.inject.spi.WithAnnotations;
+
+import io.helidon.common.configurable.ScheduledThreadPoolSupplier;
+
+import com.cronutils.descriptor.CronDescriptor;
+import com.cronutils.model.Cron;
+import com.cronutils.model.definition.CronDefinition;
+import com.cronutils.model.definition.CronDefinitionBuilder;
+import com.cronutils.model.time.ExecutionTime;
+import com.cronutils.parser.CronParser;
+
+import static com.cronutils.model.CronType.QUARTZ;
+import static javax.interceptor.Interceptor.Priority.PLATFORM_AFTER;
+
+/**
+ * Scheduling CDI Extension.
+ */
+public class SchedulingCdiExtension implements Extension {
+    private static final Logger LOGGER = Logger.getLogger(SchedulingCdiExtension.class.getName());
+    private final Map<Class<?>, AnnotatedMethod<?>> methods = new HashMap<>();
+    private final Map<AnnotatedMethod<?>, Bean<?>> beans = new HashMap<>();
+
+    private void registerMethods(
+            @Observes
+            @WithAnnotations({Scheduled.class}) ProcessAnnotatedType<?> pat) {
+        // Lookup channel methods
+        pat.getAnnotatedType().getMethods()
+                .stream()
+                .filter(am -> am.isAnnotationPresent(Scheduled.class))
+                .forEach(annotatedMethod -> {
+                    methods.put(annotatedMethod.getDeclaringType().getJavaClass(), annotatedMethod);
+                });
+    }
+
+    private void onProcessBean(@Observes ProcessManagedBean<?> event) {
+        // Gather bean references
+        Bean<?> bean = event.getBean();
+        Optional.ofNullable(methods.get(bean.getBeanClass()))
+                .ifPresent(m -> beans.put(m, bean));
+    }
+
+    private void deploymentValidation(@Observes AfterDeploymentValidation event) {
+    }
+
+    private void invoke(@Observes @Priority(PLATFORM_AFTER + 102) @Initialized(ApplicationScoped.class) Object event,
+                        BeanManager beanManager) {
+        // kickoff
+        ScheduledThreadPoolSupplier scheduledThreadPoolSupplier = ScheduledThreadPoolSupplier.builder()
```

Comment:
This must be configurable through config

Reply:
it is now